firmware/application/external/rook/ui_rook.cpp

```diff
@@ -0,0 +1,269 @@
+            text_payload.set("parent not available");
+        }
+    }
+ 
+    // `on_tx_progress` method: Updates the progress bar based on transmission progress.
+    void ROOKAppView::on_tx_progress(const uint32_t progress, const bool done) {
+        progressBar_progress.set_value(progress);  // Update progress bar value
+        if (done) {
+            stop_tx();  // Stop transmission when progress reaches maximum
+        }
+    }
+ 
+    // `draw_waveform` method: Draws the waveform on the UI based on the payload data
+    void ROOKAppView::draw_waveform() {
+        // Padding reason:
+        // In real-world scenarios, the signal would always start low and return low after turning off the radio.
+        // `waveform_buffer` only controls drawing; the actual send logic is handled by frame_fragments.
+        size_t length = payload.length();
+ 
+        // Ensure waveform length does not exceed buffer size
+        if (length + (PADDING_LEFT + PADDING_RIGHT) >= WAVEFORM_BUFFER_SIZE) {
+            length = WAVEFORM_BUFFER_SIZE - (PADDING_LEFT + PADDING_RIGHT);
+        }
+ 
+        // Left padding
+        for (size_t i = 0; i < PADDING_LEFT; i++) {
+            waveform_buffer[i] = 0;
+        }
+ 
+        // Draw the actual waveform
+        for (size_t n = 0; n < length; n++) {
+            waveform_buffer[n + PADDING_LEFT] = (payload[n] == '0') ? 0 : 1;
+        }
+ 
+        // Right padding
+        for (size_t i = length + PADDING_LEFT; i < WAVEFORM_BUFFER_SIZE; i++) {
+            waveform_buffer[i] = 0;
+        }
+ 
+        waveform.set_length(length + PADDING_LEFT + PADDING_RIGHT);
+        waveform.set_dirty();
+    }
+ 
+    // Destructor for `ROOKAppView`: Disables the transmitter and shuts down the baseband
+    ROOKAppView::~ROOKAppView() {
+        transmitter_model.disable();
+        baseband::shutdown();
+    }
+ 
+    // Constructor for `ROOKAppView`: Sets up the app view and initializes UI elements
+    ROOKAppView::ROOKAppView(NavigationView &nav)
+    : nav_{nav} {
+        baseband::run_image(portapack::spi_flash::image_tag_ook);
+      
+        add_children({
+            &field_frequency,
+            &tx_view,
+            &button_start,
+            &sample_rate,
+            &step_symbol_rate,
+            &cant_step_symbol_rate,
+            &field_sample_rate,
+            &field_symbol_rate,
+            &field_symbol_us_rate,
+            &cant_symbol_rate,
+            &text_payload,
+            &button_set,
+            &progressBar_progress,
+            &repeat,
+            &cant_repeat,
+            &field_pause_symbol,
+            &cant_pause_symbol,
+            &label_waveform,
+            &waveform,
+            &button_open,
+        });
+ 
+        // Initialize default values for controls
+        cant_pause_symbol.set_value(100);
+        cant_repeat.set_value(4);
+ 
+        button_open.on_select = [this](Button&) {
+            auto open_view = nav_.push<FileLoadView>(".TXT");
```

Comment:
It would be nice if you can:
1. add a ROOK payload folder location path in the file_path.cpp and hpp
2. Push dir here to default open the payload dir
3. write some payload example in the /sdcard/ROOK folder

These are just suggestions. you are free to choose accept or not

Reply:
- I enforce the dedicated directory. ROOK is a bit strange, why not "OOK-SAMPLES" or something talking.
- It needs dedicated extension too, like .OOK instead of .TXT, as  it would allow to launch the right app from fileman directly

Reply:
Added functionality to load the /ROOK path when loading configuration files and Replaced .TXT extension for configuration files with a new one called .ROOK